sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -1108,3 +1108,190 @@ def test_newton_solver_verbosity(capsys, verbose):
+            [0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 53.0],
+            [0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 155.0],
+            [0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 240.0],
+            [0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1635.0],
+            [0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 24.0],
+            [0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 78.0],
+            [0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 121.0],
+            [0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 692.0],
+            [0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 7.0],
+            [0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 29.0],
+            [0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 43.0],
+            [0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 245.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 20.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 33.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 40.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 316.0],
+            [0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 31.0],
+            [0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 81.0],
+            [0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 122.0],
+            [0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 724.0],
+            [0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 18.0],
+            [0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 39.0],
+            [0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 68.0],
+            [0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 344.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 3.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 16.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 25.0],
+            [0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 114.0],
+        ]
+        y = [
+            38,
+            35,
+            20,
+            156,
+            63,
+            84,
+            89,
+            400,
+            19,
+            52,
+            74,
+            233,
+            4,
+            18,
+            19,
+            77,
+            22,
+            19,
+            22,
+            87,
+            25,
+            51,
+            49,
+            290,
+            14,
+            46,
+            39,
+            143,
+            4,
+            15,
+            12,
+            53,
+            5,
+            11,
+            10,
+            67,
+            10,
+            24,
+            37,
+            187,
+            8,
+            19,
+            24,
+            101,
+            3,
+            2,
+            8,
+            37,
+            2,
+            5,
+            4,
+            36,
+            7,
+            10,
+            22,
+            102,
+            5,
+            7,
+            16,
+            63,
+            0,
+            6,
+            8,
+            33,
+        ]
```

Comment:
Isn't it possible to synthetically generate the data for a reproducer? For instance, one-hot encoding a random integer variable with 9 possible values to get the first 9 columns and then concatenating a random numerical feature with a long tail positive distribution (e.g. log normal or similar with a large enough scale parameter)?

y (or `log(y + eps)`) and the last numerical column probably need to be correlated to trigger the problem.

Alternatively, we could try to cut this dataset iteratively in half and see if the problem is still there to attempt to make it more minimal.

But if it's too challenging to find a more minimal reproducer, then so be it